cmd/calyptia/create_core_instance_aws.go

```diff
@@ -1,19 +1,206 @@
 package main
 
 import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/sethvargo/go-retry"
+
 	"errors"
 
 	"github.com/spf13/cobra"
+
+	"github.com/calyptia/api/types"
+	awsclient "github.com/calyptia/cli/aws"
+)
+
+const (
+	coreInstanceUpCheckTimeout = 10 * time.Minute
+	coreInstanceUpCheckBackoff = 5 * time.Second
+)
+
+var (
+	errCoreInstanceNotFound        = errors.New("core instance not found")
+	errCoreInstanceNotRunning      = errors.New("core instance not in running status")
+	coreInstanceUpCheckMaxDuration = func() retry.Backoff {
+		return retry.WithMaxDuration(coreInstanceUpCheckTimeout, retry.NewConstant(coreInstanceUpCheckBackoff))
+	}
+)
+
+type (
+	//go:generate moq -out core_instance_poller_mock.go . CoreInstancePoller
+	CoreInstancePoller interface {
+		Ready(ctx context.Context, name string) error
+	}
+
+	DefaultCoreInstancePoller struct {
+		CoreInstancePoller
+		config *config
+	}
 )
 
-func newCmdCreateCoreInstanceOnAWS(config *config) *cobra.Command {
+func (c *DefaultCoreInstancePoller) Ready(ctx context.Context, name string) error {
+	return retry.Do(ctx, coreInstanceUpCheckMaxDuration(), func(ctx context.Context) error {
+		instances, err := c.config.cloud.Aggregators(ctx, c.config.projectID, types.AggregatorsParams{
+			Name: &name,
+		})
+
+		if err != nil || len(instances.Items) == 0 {
+			return retry.RetryableError(err)
+		}
+
+		instance := instances.Items[0]
+		if instance.Status != types.AggregatorStatusRunning {
+			return retry.RetryableError(errCoreInstanceNotRunning)
+		}
+		return nil
+	})
+}
+
+func newCmdCreateCoreInstanceOnAWS(config *config, client awsclient.Client, poller CoreInstancePoller) *cobra.Command {
+	var (
+		tags                   []string
+		noHealthCheckPipeline  bool
+		noElasticIPv4Address   bool
+		coreInstanceVersion    string
+		coreInstanceName       string
+		environmentKey         string
+		credentials            string
+		profileFile            string
+		profileName            string
+		region                 string
+		subnetID               string
+		keyPairName            string
+		instanceTypeName       string
+		securityGroupName      string
+		elasticIPv4Address     string
+		elasticIPv4AddressPool string
+	)
+
 	cmd := &cobra.Command{
 		Use:     "aws",
 		Aliases: []string{"ec2", "amazon"},
-		Short:   "Setup a new core instance on Amazon EC2 (TODO)",
+		Short:   "Setup a new core instance on Amazon EC2",
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return errors.New("not implemented")
+			var (
+				awsInstance awsclient.CreatedInstance
+				err         error
+			)
+
+			ctx := context.Background()
+
+			if client == nil {
+				client, err = awsclient.New(ctx, coreInstanceName, region, credentials, profileFile, profileName)
+				if err != nil {
+					return fmt.Errorf("could not initialize client: %w", err)
+				}
+			}
+
+			exists, err := coreInstanceNameExists(ctx, config, coreInstanceName)
+			if err != nil && !errors.Is(err, errCoreInstanceNotFound) {
```

Comment:
no need to check for nil if we r using `errors.Is`

Reply:
Yes, its required to check != nil, as  !errors.Is(err, errCoreInstanceNotFound), if err == nil , that will evaluate to true.